### DISCUSSION TITLE
PNG to WebP conversion: some files 10x larger than with `cwebp`

### DISCUSSION THREAD

**Discussion by PerplexDaniel** [SixLabors/ImageSharp#2766] [Q&A] [answered]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have verified that I am running the latest version of ImageSharp
- [X] I have verified if the problem exist in both `DEBUG` and `RELEASE` mode
- [X] I have searched [open](https://github.com/SixLabors/ImageSharp/issues) and [closed](https://github.com/SixLabors/ImageSharp/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### ImageSharp version

3.1.4

### Other ImageSharp packages and versions

N/A

### Environment (Operating system, version and so on)

Windows 11 23H2 Build 22631.3737

### .NET Framework version

8.0

### Description

When converting some PNG files to WebP using ImageSharp the output file is way larger than when using [`cwebp`](https://developers.google.com/speed/webp/download) 1.4.0 (Google's WebP encoder).

For example, when the attached 9.01 MB PNG file is converted to WebP using ImageSharp the output image is 7.14 MB whereas `cwebp` creates an image of 392 kB, 18x smaller.

Setting a different quality on `WebpEncoder` does not seem to have any effect either which is strange.

I have seen this happen to random other PNG files too but obviously cannot conclude it happens to _every_ PNG image out there.

### Steps to Reproduce

1. Store the image as `img.png` on your computer
2. Create a console app (`dotnet new console -n ImageSharpTest`):
3. `cd ImageSharpTest`
4. `dotnet add package SixLabors.ImageSharp`
5. Copy this code into Program.cs:
```csharp
using SixLabors.ImageSharp;

Directory.SetCurrentDirectory("D:\\"); // Change to directory holding `img.png`
using Image image = Image.Load("img.png");
image.SaveAsWebp("img.webp");
```
6. `dotnet run`
7. Observe `img.webp` has a file size of 7.14 MB.

If you downloaded **Precompiled WebP utilities and library** from [Google](https://developers.google.com/speed/webp/download), you can compare this to `cwebp` as follows:

1. `cwebp <path-to-img.png> -o img.webp`
2. Observe this yields an image of roughly 392 kB, that's 18x smaller.

### Images

![img](https://github.com/SixLabors/ImageSharp/assets/24297947/672460ce-4b0e-4be6-8da9-0ede0d498a2e)

**Comment by JimBobSquarePants** [comment, marked as answer]
@PerplexDaniel I thinkt his would have been better suited as a discussion. 

It's likely cwep is using lossy encoding. 

I can use the default lossy encoding options with the following code and return a 387Kb image.

```c#
using var img = Image.Load(@"C:\Users\james\Downloads\2765.png");

WebpEncoder encoder = new ()
{
    FileFormat = WebpFileFormatType.Lossy
};
img.SaveAsWebp($@"C:\Users\james\Downloads\2765.webp", encoder);

```

**Comment by PerplexDaniel** [comment]
@JimBobSquarePants Thank you for the super fast response. 

Is there any way to configure Lossy as a default? This is used in the context of Umbraco CMS if that matters.

> I thinkt his would have been better suited as a discussion.

Apologies, it felt like a bug compared to `cwebp` but I didn't know about the `FileFormat` setting. Feel free to move / close this of course.

**Comment by JimBobSquarePants** [comment]
I don't exactly know how Umbraco sets things up but it's likely not using a custom `Configuration` instance so using  `Configuration.Default`. It's possible to update the default instance by registering your own encoders. 

You'd put this somewhere in startup.

``` c#
 Configuration.Default.ImageFormatsManager.SetEncoder(WebpFormat.Instance, new WebpEncoder() { FileFormat = WebpFileFormatType.Lossy });
```

**Comment by PerplexDaniel** [comment]
It seems this behavior has changed between 2.1.8 and 3.1.4, presumably 2.x to 3.x. 

If I run the same code (without passing the encoder to SaveAsWebP) in 2.1.8 I get a ~400KB file. Updating to 3.1.4 it is 7MB. So presumably in 2.x the default was actually `Lossy` but in 3.x it is now `Lossless`.

```csharp
using var img = Image.Load("img.png");
img.SaveAsWebp("img.webp");

// img.webp size:
// v2.1.8: 404 KB
// v3.1.4: 7.14 MB
```

**Reply by JimBobSquarePants** [reply]
We default to Lossless so that no image data is lost with default conversion. 9MB PNGs aren’t all that common

**Reply by PerplexDaniel** [reply]
That sounds sensible. We were not aware of the change in default behavior from v2 to v3 though so it came as a bit of a surprise when upgrading a project. I will ask Umbraco if they are willing to configure the WebP encoder default as Lossy so it behaves like in previous versions of the CMS (and ImageSharp) but we can already apply this to our project so for us it's good now.
